nb/ide.launcher/unix/netbeans

```diff
@@ -222,8 +222,29 @@ case "`uname`" in
         # information."
         # https://userbase.kde.org/KDE_System_Administration/Environment_Variables
         if [ ! -z "$KDE_FULL_SESSION" ] ; then
-            echo "Detected KDE; adding awt.useSystemAAFontSettings=on"
-            extra_automatic_options="-J-Dawt.useSystemAAFontSettings=on"
+            # Try to detect the correct subpixel antialiasing mode
+            # (https://github.com/apache/netbeans/issues/4228)
+
+            # See https://docs.gtk.org/gtk4/property.Settings.gtk-xft-rgba.html
+            # See https://docs.oracle.com/javase/7/docs/technotes/guides/2d/flags.html#aaFonts
+            case "`command xrdb -query 2> /dev/null | grep Xft.rgba | cut -d ':' -f2 | xargs`" in
+                rgb)
+                    extra_automatic_options="-J-Dawt.useSystemAAFontSettings=lcd_hrgb"
+                    ;;
+                bgr)
+                    extra_automatic_options="-J-Dawt.useSystemAAFontSettings=lcd_hbgr"
+                    ;;
+                vrgb)
+                    extra_automatic_options="-J-Dawt.useSystemAAFontSettings=lcd_vrgb"
+                    ;;
+                vbgr)
+                    extra_automatic_options="-J-Dawt.useSystemAAFontSettings=lcd_vbgr"
+                    ;;
+                *)
+                    extra_automatic_options="-J-Dawt.useSystemAAFontSettings=on"
+                    ;;
+            esac
+            echo "Detected KDE; use explicit setting for font antialiasing ($extra_automatic_options)"
```

Comment:
do I understand this issue correctly that:
 - JVM doesn't enable AA at all on some KDE systems, thats why NB began to set `-J-Dawt.useSystemAAFontSettings=on` only on KDE
 - on other KDE systems, `-J-Dawt.useSystemAAFontSettings=on` would pick the wrong mode, so the JVM does need even more help there, so we end up to configure everything font AA related on KDE

If this works - great. However this probably should be upstreamed to OpenJDK if possible, looks like mode probing on KDE is lacking a bit behind. The Java 6 doc already mentioned that KDE won't be detected, interesting that this didn't change since then :)

Reply:
* First bullet: Correct.
* Second bullet: On these "other KDE systems", the JDK (reportedly) detects the subpixel antialiasing setting just fine. But the "on" option for awt.useSystemAAFontSettings actually refers specifically to the greyscale-type antialiasing. Which is a safe but not fully optimal setting.

> However this probably should be upstreamed to OpenJDK if possible, looks like mode probing on KDE is lacking a bit behind.

Yes, it's really an old JDK deficiency. Once NetBeans has run with this patch for a while, I can submit a JDK bug reporting our experience...

Reply:
> Yes, it's really an old JDK deficiency. Once NetBeans has run with this patch for a while, I can submit a JDK bug reporting our experience...

awesome!

Reply:
(I also did one more experiment just now, confirming that the JDK does _not_ by itself detect AA settings just because Xft.rgba is present. So I don't know the exact conditions in which the JDK would be able to detect AA settings by itself on KDE, other than NickLion's report from openSUSE Tumbleweed with KDE Plasma.)

Reply:
I updated the comment to change "Java/AWT/Swing will correctly detect text anti-aliasing settings on GNOME, but not on KDE" to "not (always) on KDE".